# KT-Group-mNGS-Scripts

A shared repository for internal use within the **KT Group**, hosting commonly used scripts for metagenomic analysis.

---

## ğŸ“š Module Overview

| Module                    | Description                                       | Link |
|---------------------------|---------------------------------------------------|------|
| **ğŸ§© Run Kraken2 and Bracken** | Run Kraken2 and Bracken (KB) on raw Nanopore data (including *removal of human reads*) |ğŸ‘‰ [View](https://github.com/Sinceter/KT-Group-mNGS-Scripts/blob/main/Run_Bracken_and_Kraken2/README.md) |
| **ğŸ“ Kraken2 Report Summary**| Summarize Kraken2-style reports and generate easy-to-read table     |ğŸ‘‰ [View](https://github.com/Sinceter/Microbial-Clinical-Report?tab=readme-ov-file#kraken2-style-reports-summary) |
| **ğŸ§¬ De novo Assembly**      | Genome assembly for long/short reads     |ğŸ‘‰ [View](https://github.com/Sinceter/KT-Group-mNGS-Scripts/blob/main/De_novo_assembly/README.md) |
| **Generate_consensus_for_CHIKV/Zika/etc.** | Generate consensus sequence using Medaka (for Nanopore reads) |ğŸ‘‰ğŸ¼[View](https://github.com/Sinceter/KT-Group-mNGS-Scripts/blob/main/Generate_consensus_medaka/README.md) |
| **Generate_consensus_for_RSV** | Generate RSV consensus using ARTIC+IRMA pipeline (for Nanopore reads) |ğŸ‘‰ğŸ¼[View](https://github.com/Sinceter/KT-Group-mNGS-Scripts/blob/main/Generate_consensus_RSV_ARTIC%2BIRMA/README.md) |
| **Nanopore reads mapping script** | Mapping Nanopore reads to a reference genome using minimap2 | ğŸ‘‰ğŸ¼[View](https://github.com/Sinceter/KT-Group-mNGS-Scripts/blob/main/Nanopore_reads_mapping/README.md) |

---


## Contact

For questions, bug reports, or suggestions, please contact **liaixin06@connect.hku.hk** or open an issue on the [GitHub repository](https://github.com/Sinceter/KT-Group-mNGS-Scripts).

