# De novo assembly scripts
The scripts are stored in CVVT server.

> [!NOTE]
> The terminologies used here:   </br>
> **Short reads** → sequencing reads from **Illumina / MGI** platforms (usually < 300 bp). </br>
> **Long reads** → sequencing reads from **Nanopore / PacBio** platforms (can reach kb scale).  


# 1. Copy target script to your working directory
## Script for short reads assembly with paired-end reads
- Software: Megahit
```bash
cp /home/kelvinto/kelvinto/aixin/00.Data/06.assembly/run.megahit.for.short.reads.sh <your working directory>
```
> Please note that even though Megahit has been designed for short reads assembly, the script ```run.megahit.for.long.reads-try.sh``` can still be used in following two conditions:
> 1. Single-end reads generated by Illumina / MGI platform;
> 2. Single-end reads generated by Nanopore / PacBio platform, only for personal trial. </br>

> Use command:

```
cp /home/kelvinto/kelvinto/aixin/00.Data/06.assembly/run.megahit.for.long.reads-try.sh <your working directory>
```


## Script for long reads assembly with single-end reads
- Software: Flye
```bash
cp /home/kelvinto/kelvinto/aixin/00.Data/06.assembly/run.flye.sh <your working directory>
```

# 2. Usage
```
bash <target script> <fastq file>
```

# 3. Output
## Megahit output folder
```
megahit_${prefix_name_of_your_fastq_file}_${date}
```
The assembled contigs from Megahit are stored in:
```
megahit_${prefix_name_of_your_fastq_file}_${date}/final.contigs.fa
```
## Flye output folder
```
flye_${prefix_name_of_your_fastq_file}_${date}
```
The assembled contigs from Flye are stored in:
```
flye_${prefix_name_of_your_fastq_file}_${date}/assembly.fasta
```
