# Consensus Generation Pipeline

## Description
This pipeline processes Nanopore sequencing data to generate consensus sequences for any species as long as you provide the reference genome. It uses Medaka for consensus calling, masks low-coverage regions, and outputs the final consensus sequences.

---

## Parameters
The pipeline accepts the following parameters:

- **`-i`**: Input FASTQ file (compressed, required).
- **`-r`**: Reference genome in FASTA format (required).
- **`-c`**: Coverage threshold for masking low-coverage regions (optional, default: `10`).
  - Please see example below.
    - When `-c` is set to 10, **all 3 consensus segments of Barcode74 will be masked**, so no consensus sequences will be generated for Barcode74 because each segment has at most 8 supporting reads.
    - When `-c` is 5, **1 consensus sequence will be generated for Barcode74**.
  - <img width="1279" height="503" alt="image" src="https://github.com/user-attachments/assets/ad8b5fb1-52c8-4236-8c4e-a0c8cae9f568" />

- **`-t`**: Threads (optional, default: `8`).
- **`-m`**: Medaka model to use (optional, default: `r1041_e82_400bps_sup_v5.0.0`).
  - You can check [Medaka GitHub #Model](https://github.com/nanoporetech/medaka?tab=readme-ov-file#models) for models that can be used.

---

## Usage
Run the pipeline for a single sample:
```bash
bash run_medaka_consensus.sh -i sample.fastq.gz -r ChikV_ref_sequence_PV593524.fasta -c 5 -t 12
```

---

## Output
```
medaka_Filter_len800-1800bp_trimmed_barcode74_20250912-1150/ # output folder with time stamp
├── calls_to_draft.bam # bam file showing read alignments to the reference
├── calls_to_draft.bam.bai # index file
├── consensus.fasta # original segmented consensus sequences generated by medaka 
├── consensus.fastq # original segmented consensus sequences generated by medaka 
├── consensus.masked.fasta # FINAL consensus sequence with low-coverage regions marked as 'N' (coverage criteria use the number you set: `-c` [INT])
├── consensus_probs.hdf # base-level probability file generated by Medaka (HDF5 format)
├── consensus.stitch.fasta # segmented consensus sequences stitched into one sequence according to the reference genome coordinates, gaps filled with 'N'
└── lowcov.bed # low-coverage regions (masked as 'N' in the final consensus)

0 directories, 8 files
```
**Please use consensus.masked.fasta as final consensus sequence:**
<img width="892" height="232" alt="image" src="https://github.com/user-attachments/assets/af55f0ee-a52b-43e4-98cc-858a4c4cec67" />

---

## Where to download the scripts
The scripts are stored in the right computer in the sequencing room (IP: 10.64.148.20), you can copy this to your working directory and run the script according to [usage](#usage).
```
cp /home/kt_jdip/aixin/11.Chikungunya.PCR/consensus/run_medaka_consensus.sh <your working directory>
```

---

## Example

### CHIKV
- The consensus sequences of Chikungunya virus (generated at minimum 10 reads coverage) is located in `/home/kt_jdip/aixin/11.Chikungunya.PCR/consensus/filter_min-10-reads_to_generate_consensus/summary`.
- Overview
<img width="1163" height="723" alt="image" src="https://github.com/user-attachments/assets/2b1106ff-25a7-4ac7-bfc9-c57cd3cafb9a" />


