# Consensus Generation Pipeline
## Update: 2025-10-28
## Content
- [Description](#description)
- [Usage](#usage)
- [Parameters](#parameters)
- [Output](#output)
- [Example](#example)
  - [CHIKV (Nanopore20250826)](#chikv-nanopore20250826)
  - [CHIKV (Nanopore20251024)](#chikv-nanopore20251024)
  - [Zika virus (Nanopore20251009_2)](#zika-virus-nanopore20251009_2)

## Description
Based on Nanopore sequencing data, this pipeline will do:
1. **Trimming** — remove primer or adapter regions from both ends of reads (optional).  
2. **Length filtering** — retain reads within the defined size window (optional).  
3. **Medaka consensus calling** — generate polished consensus sequence.  
4. **Coverage masking** — mask bases below coverage threshold with `N` and outputs the final consensus sequences. 

> [!WARNING]
> You can run the pipeline only if you can access to the right computer in the sequencing room (IP: 10.64.148.20) (KT group). 

---

## Usage
Run the pipeline for a single sample:
```bash
cd <your working directory>
SCRIPT="/home/kt_jdip/aixin/11.Chikungunya.PCR/consensus/medaka_consensus_pipeline.sh"

# Examples:
## default trim 30 bp + filter length 800-1800
  bash $SCRIPT -i sample.fastq.gz -r ref.fasta

## trim 20 bp + filter length 700-2000
  bash $SCRIPT -i sample.fastq.gz -r ref.fasta -b 20 -e 20 -L 700 -U 2000

## skip primer trimming and reads length filtering
  bash $SCRIPT -i sample.fastq.gz -r ref.fasta --no-trim --no-filter-length -t 16 

```

---
## Parameters
The pipeline accepts the following parameters:

### Required input files

| **Parameter** | **Type / Default** | **Description** | **Example** |
|----------------|--------------------|-----------------|--------------|
| `-i <file>` | *Required* | Input FASTQ file (`.fastq` or `.fastq.gz`). | `-i sample.fastq.gz` |
| `-r <file>` | *Required* | Reference genome in FASTA format, used for alignment and consensus polishing. | `-r ChikV_ref.fasta` |
| *(Output directory)* | auto | Results are saved in a folder named `medaka_pipeline_<sample>`. | `medaka_pipeline_nanopore2xxxbarcode01` |

---

### Preprocessing: trimming

| **Parameter** | **Default** | **Description** | **Example** |
|----------------|-------------|-----------------|--------------|
| `-b <int>` | `30` | Number of bases trimmed from the **beginning** of each read. | `-b 20` |
| `-e <int>` | `30` | Number of bases trimmed from the **end** of each read. | `-e 20` |
| *(if you do not want to trim reads, use parameter below:)* | - | - | — |
| `--no-trim` |   | Disable trimming step entirely. | `--no-trim` |


---

### Preprocessing: length filtering

| **Parameter** | **Default** | **Description** | **Example** |
|----------------|-------------|-----------------|--------------|
| `-L <int>` | `800` | Minimum read length to retain. | `-L 700` |
| `-U <int>` | `1800` | Maximum read length to retain. | `-U 2000` |
| *(if you do not want to filter length, use parameter below:)* | — | - | — |
| `--no-filter-length` |  | Disable length filtering step. | `--no-filter-length` |

---

### Medaka options

| **Parameter** | **Default** | **Description** | **Example** |
|----------------|-------------|-----------------|--------------|
| `-t <int>` | `12` | Number of CPU threads for **Medaka** and **pigz** compression. | `-t 16` |
| `-m <str>` | `r1041_e82_400bps_sup_v5.0.0` | Medaka model name (depends on basecaller version and read type). | `-m r1041_e82_400bps_fast_v5.0.0` |
| `-c <int>` | `10` | Coverage threshold for low-coverage masking, please note that bases with coverage less than the number you specified with `-c` will be replaced by `N`. | `-c 20` |
> [!NOTE]
> For -m, you can check [Medaka GitHub #Model](https://github.com/nanoporetech/medaka?tab=readme-ov-file#models) for models that can be used.
> 
> For -c, please see example below.
>
>   - When `-c` is set to 10, *all 3 consensus segments of Barcode74 will be masked*, so no consensus sequences will be generated for Barcode74 because each segment has at most 8 supporting reads.
>
>   - When `-c` is 5, *only 1 consensus segment (the region around 10kb-11kb in the figure below) will be generated for Barcode74*.
>
> <img width="1279" height="503" alt="image" src="https://github.com/user-attachments/assets/ad8b5fb1-52c8-4236-8c4e-a0c8cae9f568" />


---


## Output
```
medaka_pipeline_barcode84/ ---> # output folder with sample identifier
├── calls_to_draft.bam -----------------------------------------> # bam file showing read alignments to the reference
├── calls_to_draft.bam.bai -------------------------------------> # index file
├── consensus.fasta --------------------------------------------> # original segmented consensus sequences generated by medaka 
├── consensus.fastq --------------------------------------------> # original segmented consensus sequences generated by medaka 
├── consensus.masked.fasta -------------------------------------> # FINAL consensus sequence 
├── barcode84_medaka.consensus.masked.fasta --------------------> # FINAL consensus sequence with sample name
├── consensus_probs.hdf ----------------------------------------> # base-level probability file generated by Medaka (HDF5 format)
├── consensus.stitch.fasta -------------------------------------> # segmented consensus sequences stitched into one sequence according to the reference genome coordinates, gaps filled with 'N'
├── Filter_len800-1800bp_barcode84.fastq.gz --------------------> # if produced, this is the reads file after length filtering
├── lowcov.bed -------------------------------------------------> # low-coverage regions (masked as 'N' in the final consensus)
└── Trimmed_barcode89.fastq.gz ---------------------------------> # if produced, this is the reads file after reads primer trimming

0 directories, 9 files
```

> [!NOTE]
> **The final consensus sequence is `consensus.masked.fasta`.**
> If it is empty, then no consensus sequence can be generated based on this sample
<img width="892" height="232" alt="image" src="https://github.com/user-attachments/assets/af55f0ee-a52b-43e4-98cc-858a4c4cec67" />


---

## Example

### CHIKV (Nanopore20250826)
- The reference genome of CHIKV is located in `/home/kt_jdip/aixin/11.Chikungunya.PCR/consensus/ChikV_ref_sequence_PV593524.fasta`
- The consensus sequences of Chikungunya virus (generated at minimum 10 reads coverage) is located in `/home/kt_jdip/aixin/11.Chikungunya.PCR/consensus/filter_min-10-reads_to_generate_consensus/summary`.
### CHIKV (Nanopore20251024)
- Result: `/home/kt_jdip/aixin/11.CHIKV.Nanopore20251024`

### Zika virus (Nanopore20251009_2)
- The reference genome of Zika virus is located in `/home/kt_jdip/aixin/12.Zika/Zika_MR766_ref.fas`
- The consensus sequences of Zika virus is `/home/kt_jdip/aixin/12.Zika/medaka_20251009_2_48hps_barcode01_20251015-1512/consensus.masked.fasta`

