# Consensus Generation Pipeline
## Content
- [Description](#description)
- [Usage](#usage)
- [Parameters](#parameters)
- [Output](#output)
- [Example](#example)
  - [CHIKV (Nanopore20250826)](#chikv-nanopore20250826)
  - [Zika virus (Nanopore20251009_2)](#zika-virus-nanopore20251009-2)

## Description
Based on Nanopore sequencing data, this pipeline uses Medaka for consensus calling, masks low-coverage regions, and outputs the final consensus sequences. 

> [!WARNING]
> You can run the pipeline only if you can access to the right computer in the sequencing room (IP: 10.64.148.20) (KT group). 

---

## Usage
Run the pipeline for a single sample:
```bash
cd <your working directory>
SCRIPT="/home/kt_jdip/aixin/11.Chikungunya.PCR/consensus/run_medaka_consensus.sh"
bash $SCRIPT -i <fastq.gz file of your sample> -r <reference genome> -c <coverage depth> -t <threads number> -m <Medaka model>
```

---
## Parameters
The pipeline accepts the following parameters:

- **`-i`**: Input FASTQ file (compressed, **required**).
- **`-r`**: Reference genome in FASTA format (**required**).
- **`-c`**: Coverage threshold for masking low-coverage regions (optional, default: `10`).
  - Please see example below.
    - When `-c` is set to 10, *all 3 consensus segments of Barcode74 will be masked*, so no consensus sequences will be generated for Barcode74 because each segment has at most 8 supporting reads.
    - When `-c` is 5, *only 1 consensus segment (the region around 10kb-11kb in the figure below) will be generated for Barcode74*.
  - <img width="1279" height="503" alt="image" src="https://github.com/user-attachments/assets/ad8b5fb1-52c8-4236-8c4e-a0c8cae9f568" />

- **`-t`**: Threads (optional, default: `8`).
- **`-m`**: Medaka model to use (optional, default: `r1041_e82_400bps_sup_v5.0.0`).
  - You can check [Medaka GitHub #Model](https://github.com/nanoporetech/medaka?tab=readme-ov-file#models) for models that can be used.

---


## Output
```
medaka_Filter_len800-1800bp_trimmed_barcode74_20250912-1150/ ---> # output folder with time stamp
├── calls_to_draft.bam -----------------------------------------> # bam file showing read alignments to the reference
├── calls_to_draft.bam.bai -------------------------------------> # index file
├── consensus.fasta --------------------------------------------> # original segmented consensus sequences generated by medaka 
├── consensus.fastq --------------------------------------------> # original segmented consensus sequences generated by medaka 
├── consensus.masked.fasta -------------------------------------> # FINAL consensus sequence with low-coverage regions marked as 'N' (coverage criteria use the number you set: `-c` [INT])
├── consensus_probs.hdf ----------------------------------------> # base-level probability file generated by Medaka (HDF5 format)
├── consensus.stitch.fasta -------------------------------------> # segmented consensus sequences stitched into one sequence according to the reference genome coordinates, gaps filled with 'N'
└── lowcov.bed -------------------------------------------------> # low-coverage regions (masked as 'N' in the final consensus)

0 directories, 8 files
```

> [!NOTE]
> **The final consensus sequence is `consensus.masked.fasta`.**
> If it is empty, then no consensus sequence can be generated based on this sample
<img width="892" height="232" alt="image" src="https://github.com/user-attachments/assets/af55f0ee-a52b-43e4-98cc-858a4c4cec67" />


---

## Example

### CHIKV (Nanopore20250826)
- The reference genome of CHIKV is located in `/home/kt_jdip/aixin/11.Chikungunya.PCR/consensus/ChikV_ref_sequence_PV593524.fasta`
- The consensus sequences of Chikungunya virus (generated at minimum 10 reads coverage) is located in `/home/kt_jdip/aixin/11.Chikungunya.PCR/consensus/filter_min-10-reads_to_generate_consensus/summary`.

### Zika virus (Nanopore20251009_2)
- The reference genome of Zika virus is located in `/home/kt_jdip/aixin/12.Zika/Zika_MR766_ref.fas`
- The consensus sequences of Zika virus is `/home/kt_jdip/aixin/12.Zika/medaka_20251009_2_48hps_barcode01_20251015-1512/consensus.masked.fasta`

